lib/api/BackbeatAPI.js

```diff
@@ -76,11 +76,12 @@ class BackbeatAPI {
         this._backbeatMetadataProxy =
             new BackbeatMetadataProxy(this._repConfig.source);
         // Redis instance is used for getting/setting keys
-        this._redisClient = new RedisClient(this._redisConfig, this._logger);
+        this._redis = new RedisClient(this._redisConfig, this._logger);
         // Redis expiry increased by an additional interval so we can reference
         // the immediate older data for average throughput calculation
-        this._redis = new RedisClient(this._localCacheRedis, this._logger);
-        this._statsClient = new StatsModel(this._redis, INTERVAL,
+        this._redisClient = new RedisClient(this._localCacheRedis,
```

Comment:
This will set failed keys in the local cache. Unless I'm mistaken, I think we want to use Redis HA for failed keys.